R/summarize_weighted_production.R

```diff
@@ -282,7 +286,11 @@ add_technology_share_target <- function(data) {
       .data$name_ald,
       .data$region
     ) %>%
-    mutate(technology_share_target = .data$production_target / sum(.data$production_target)) %>%
+    mutate(
+      .x = sum(.data$production_target),
+      technology_share_target = ifelse(.data$.x == 0, 0, .data$production_target / .data$.x),
+      .x = NULL
+    ) %>%
```

Comment:
Just an idea, I wonder if we could extract this pattern into a wrapper around `mutate()` with a new `.x` argument that will be automatically destroyed  after use. Nothing to take action but to spark thoughts.

Reply:
Ya this seems like an interesting idea. Would have to think carefully about what to call the new function